src/azure-cli/azure/cli/command_modules/profile/_help.py

```diff
@@ -56,39 +64,67 @@
 helps['account show'] = """
 type: command
 short-summary: Get the details of a subscription.
-long-summary: If no subscription is specified, shows the current subscription.
+long-summary: If no subscription is specified, information for the current subscription is returned.
+examples:
+- name: Get the current default subscription returning results in your default format.
+  text: az account show
+- name: Get information about a named subscription returning results as a table. Use `--output yaml` for plain text.
+  text: az account show --name "My subscription name" --output table
+- name: Get information about a subscription using the subscription ID.
+  text: az account show --name MySubscriptionID
+- name: >
+    Get specific information about a subscription using `--query`.
+    To see available query options, first run `az account show --output json`.
+    For in-depth `--query` examples see [How to query Azure CLI command output using a JMESPath query](https://learn.microsoft.com/cli/azure/query-azure-cli).
+  text: >
+    az account show --query user.name
+- name: Return results in plain text.
+  text: az account show --query MyTenantID --output tsv
+- name: Store the default subscription ID in a variable.
+  text: >
+    # Bash script
+    subscriptionId="$(az account show --query id --output tsv)"
+    echo "The current subscription ID is $subscriptionId"
 """
 
 helps['account set'] = """
 type: command
-short-summary: Set a subscription to be the current active subscription.
-"""
-
-helps['account show'] = """
-type: command
-short-summary: Get the details of a subscription.
-long-summary: If the subscription isn't specified, shows the details of the default subscription.
+short-summary: Change the active subscription.
+examples:
+- name: Change the active subscription using a subscription name
+  text: az account set --subscription "My subscription name"
+- name: Change the active subscription using a subscription ID
+  text: az account set --subscription MySubscriptionID
+- name: Change the active subscription using a variable.
+  text: >
+    # Bash script
+    subscriptionId="$(az account list --query "[?contains(name, 'OneOfMySubscriptionNames')].id" --output tsv)"
+    az account set --subscription $subscriptionId
+    echo "The current subscription has been set to $subscriptionId"
```

Comment:
Similar to https://github.com/Azure/azure-cli/pull/24332#discussion_r1003915870

These lines can simply be done with

```
az account set --subscription OneOfMySubscriptionNames
```